File: Java/src/main/java/edu/cmu/cs/cs214/rec04/InheritanceSortedIntList.java
package edu.cmu.cs.cs214.rec04;

/**
 * InheritanceSortedIntList -- a variant of a SortedIntList that keeps
 * count of the number of attempted element insertions (not to be confused
 * with the current size, which goes down when an element is removed)
 * and exports an accessor (totalAdded) for this count.
 *
 * @author Nora Shoemaker
 *
 */

public class InheritanceSortedIntList extends SortedIntList{
    // Write your implementation below with API documentation

    private int addCount = 0;

    public InheritanceSortedIntList() {
        super();
    }

    @Override
    public  boolean add(int num) {
        addCount += 1;
        return super.add(num);
    }

    public int getTotalAdded() {
        return addCount;
    }
}

File: Typescript/src/InheritanceSortedIntList.ts
import { SortedIntList } from './hidden/SortedIntListLibrary.js'

/**
 * InheritanceSortedIntList -- a variant of a SortedIntList that keeps
 * count of the number of attempted element insertions (not to be confused
 * with the current size, which goes down when an element is removed)
 * and exports an accessor (totalAdded) for this count.
 *
 * @author Nora Shoemaker
 *
 */

class InheritanceSortedIntList extends SortedIntList {
  // Write your implementation below with API documentation

  private addCount = 0

  add = (num: number): boolean => {
    this.addCount += 1
    return super.add(num)
  }

  getTotalAdded = (): number => {
    return this.addCount
  }
}

export { InheritanceSortedIntList }
